casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -105,14 +105,21 @@ object BlockAPI {
       _.withCasper[ApiErr[String]](
         casper =>
           for {
-            resultOpt <- triggerProposeF(casper, isAsync)
-            // wait till propose pipe assign proposeID to propose and resolve proposeIDDef
-            r <- resultOpt match {
-                  case Some(Right(blockHash)) =>
-                    logSucess(s"Success! Block ${blockHash.base16String} created and added.")
-                  case Some(Left(seqNum)) =>
-                    logSucess(s"Success: proposing block with seqNum ${seqNum}")
-                  case None => logDebug(s"Failure: another propose is in progress")
+            // Trigger propose
+            proposerResult <- triggerProposeF(casper, isAsync)
+            r <- proposerResult match {
+                  case ProposerEmpty =>
+                    logDebug(s"Propose failed: another propose is in progress")
+                  case ProposerFailure(status, seqNumber) =>
+                    logDebug(s"Propose failed: $status (seqNum $seqNumber)")
```

Comment:
@zsluedem should these failed statuses be in some specific format for propose script? Really, it doesn't make sense to write the same message to logs and return to API.

Reply:
> Really, it doesn't make sense to write the same message to logs and return to API.
Agree 100%. But that's how it is right now.